src/components/sidebar/sidebar.js

```diff
@@ -194,18 +214,39 @@ export const BSidebar = /*#__PURE__*/ Vue.extend({
         [h(BIconX)]
       )
       $header = right ? [$close, $title] : [$title, $close]
-      $header = h('header', { staticClass: 'b-sidebar-header', class: this.headerClass }, $header)
+      $header = h(
+        'header',
+        {
+          staticClass: `${CLASS_NAME}-header`,
+          class: this.headerClass
+        },
+        $header
+      )
+    }
+
+    const $content = h(
+      'div',
+      {
+        staticClass: `${CLASS_NAME}-content`,
+        class: this.contentClass
+      },
+      this.normalizeSlot('default', scope)
+    )
+
+    let $body = h()
+    if (!(this.lazy && !this.localShow)) {
+      $body = h('div', { staticClass: `${CLASS_NAME}-body` }, [$header, $content])

```

Comment:
Header should not be part of body... it should be outside of it.

As the header needs to be there for accessibility reason when hidden (as it acts as the accessible label for the sidebar)